python/pyfunc-server/pyfuncserver/sampler/sampler.py

```diff
@@ -0,0 +1,31 @@
+import random
+from abc import ABC, abstractmethod
+from typing import Optional
+
+class Sampler(ABC):
+    @abstractmethod
+    def should_sample(self, request_id: Optional[int] = None) -> bool:
+        pass
+
+class RatioSampling(Sampler):
```

Comment:
I used opentelemetry as reference, you can check the implementation [here](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py#L253) , I'm honestly don't know how the distribution looks like with the suggested above, that's why I used the one that adopted by the well known library

Reply:
Got it. Could you leave an inline comment with the link for reference? It's better to use the latest version [v1.21.0](https://github.com/open-telemetry/opentelemetry-python/blob/v1.21.0/opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py#L253) rather than the main branch.